handler.sh

```diff
@@ -1,38 +1,55 @@
 #
 # brew-command-not-found script for OS X
 #
-# Usage: Source it somewhere in your .bashrc
+# Usage: Source it somewhere in your .bashrc(bash) or .zshrc(zsh)
 #
 # Author: Baptiste Fontaine
 # URL: https://github.com/bfontaine/homebrew-command-not-found
 # License: MIT
 # Version: 0.2.0
-# 
+#
+
+if ! which brew > /dev/null; then return; fi
 
-[ ! -z "$(which brew)" ] && command_not_found_handle() {
+homebrew_command_not_found_handle() {
 
     local cmd="$1"
 
     # <from Linux Journal>
     #   http://www.linuxjournal.com/content/bash-command-not-found
 
-    export TEXTDOMAIN=command-not-found
-
     # do not run when inside Midnight Commander or within a Pipe
     if test -n "$MC_SID" -o ! -t 1 ; then
-        echo $"$cmd: command not found"
+        [[ "$SHELL" == */bash ]] && \
+            TEXTDOMAIN=command-not-found echo $"$cmd: command not found"
         return 127
     fi
 
     # </from Linux Journal>
 
-    local txt=$(brew which-formula --explain $cmd 2>/dev/null)
+    local txt="$(brew which-formula --explain $cmd 2>/dev/null)"
 
     if [ -z "$txt" ]; then
-        echo $"$cmd: command not found"
+        [[ "$SHELL" == */bash ]] && \
+            TEXTDOMAIN=command-not-found echo $"$cmd: command not found"
     else
         echo "$txt"
     fi
 
     return 127
 }
+
+case "$SHELL" in
```

Comment:
From what I understand, `$SHELL` is the *preferred* shell, but not the running one. For example, if I run `zsh` in `bash`, `$SHELL` will still be `…/bash`. According to [this SO answer](http://stackoverflow.com/questions/9910966/how-to-tell-if-its-using-zsh-or-bash) we could check for `$ZSH_VERSION` and `$BASH_VERSION`.